### DISCUSSION TITLE
Problems when action runs on a tag push

### DISCUSSION THREAD

**Discussion by olilarkin** [JamesIves/github-pages-deploy-action#490] [General]
Hello,

Thanks very much for your hard work on this. I've got a problem and I'm not sure the best approach to deal with it. I want to use this action on push, only with a specific git tag, but when I do that it ends up deleting my gh-pages branch and yet says deployment successful. With no tag, it works fine.

any ideas what I should do? I guess the git tag checkout is a detached head and has no notion of gh-pages branch?

thanks!

action log:

```
Reinitialized existing Git repository in /github/workspace/.git/
error: pathspec 'master' did not match any file(s) known to git.
Deploying to GitHub...
fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
To https://github.com/jamesb93/iPlug2OOS.git
 - [deleted]         gh-pages
Deployment succesful!

```

`      
on:
  push:
    tags:
      - "v*"

- name: Publish to pages
        uses: JamesIves/github-pages-deploy-action@master
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          BASE_BRANCH: master
          BRANCH: gh-pages
          FOLDER: ${{env.PROJECT_NAME}}/build-web
`

**Comment by JamesIves** [comment]
Can you try upgrading the action to the latest version? You're still running on version 1 and a lot has changed since then. What you have should be fine, just change the version number to `JamesIves/github-pages-deploy-action@3.7.1`.

**Comment by olilarkin** [comment]
ok thanks , i assumed master would include the latest changes.. i'll try changing it

**Comment by olilarkin** [comment]
With the new version I get "error branch is required". On the plus side, deployment fails now.

Should I add a step to create a branch from the detached HEAD state ?

https://github.com/iPlug2/iPlug2OOS/runs/1347871623

**Reply by JamesIves** [reply]
Swap out `env` with `with` for the variables specific to this action.

**Comment by olilarkin** [comment]
thanks, that worked :-)
